lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -1690,12 +1696,17 @@ Type *UpdateFieldTypeForLegacyLayout(Type *Ty,
 
 StructType *UpdateStructTypeForLegacyLayout(StructType *ST,
                                             DxilTypeSystem &TypeSys,
-                                            Module &M) {
+                                            Module &M,
+                                            bool includeTopLevelResource) {
   bool bUpdated = false;
   unsigned fieldsCount = ST->getNumElements();
-  std::vector<Type *> fieldTypes(fieldsCount);
+  std::vector<Type *> fieldTypes;
+  fieldTypes.reserve(fieldsCount);
```

Comment:
They aren't equivalent.  The former initializes the vector with fieldsCount empty elements, while the latter only reserves at least that much space, because I'll be copying elements and potentially omitting ones, instead of assuming a 1:1 match.